front50-core/src/main/groovy/com/netflix/spinnaker/front50/UntypedUtils.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.front50;
+
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.SneakyThrows;
+
+/** Helper class for dealing with untyped data / Groovy migration. */
+@NonnullByDefault
+public class UntypedUtils {
+
+  @Nullable
+  @SneakyThrows
+  public static Object getProperty(Object obj, String propertyName) {
+    Field f = obj.getClass().getDeclaredField(propertyName);
+    f.setAccessible(true);
+    return f.get(obj);
+  }
+
+  @SneakyThrows
+  public static void setProperty(Object obj, String propertyName, @Nullable Object value) {
+    Field f = obj.getClass().getDeclaredField(propertyName);
+    f.setAccessible(true);
+    f.set(obj, value);
+  }
+
+  public static boolean hasProperty(Object obj, String propertyName) {
+    return Arrays.stream(obj.getClass().getDeclaredFields())
+        .anyMatch(it -> it.getName().equals(propertyName));
+  }
+
+  @SneakyThrows
+  public static Map<String, String> getProperties(Object obj) {
+    Map<String, String> m = new HashMap<>();
+    Arrays.stream(obj.getClass().getDeclaredFields())
```

Comment:
this won't pick up anything from a superclass or interface (not sure that matters for how we are using it, and I can't imagine we will write any new code using this class)

Reply:
I don't know if we need the additional complexity. Hopefully we can land on something to get rid of this class in the medium-future, but as you say, I don't think we'll be extending this class any time soon.